questionpy/form/validation.py

```diff
@@ -0,0 +1,118 @@
+        else:
+            message += f" (Form contains no general element or section named '{local_name}')"
+        super().__init__(node, message)
+
+
+def _absolute_name(node: Union[_FormNode, str], *nodes: Union[_FormNode, str]) -> str:
+    name_parts = []
+    for node in (node, *nodes):
+        if isinstance(node, str):
+            name_parts.append(node)
+        elif not isinstance(node, OptionsFormDefinition):
+            name_parts.append(node.name)
+
+    return name_parts[0] + "".join(f"[{part}]" for part in name_parts[1:])
+
+
+def _resolve_reference(reference: str, referee: str, parents: Sequence[_FormNode]) -> _FormNode:
+    parents = list(parents)
+    parts = reference.replace("]", "").split("[")
+    if not parts:
+        raise FormError(referee, f"Empty reference: '{reference}'")
+
+    for i, part in enumerate(parts):
+        if part == "..":
+            parents.pop()
+            continue
+
+        children = _get_children(parents[-1])
+
+        matching_children = [child for child in children if hasattr(child, "name") and child.name == part]
+        if not matching_children:
+            # Element not found.
+            raise FormReferenceError(referee, reference, getattr(parents[-1], "name", None), part)
+        if len(matching_children) > 1:
+            # Ambiguous reference, i.e. duplicate name.
+            raise FormError(_absolute_name(*parents, part), "Duplicate element or section name")
+        matching_child = matching_children[0]
+
+        if i < len(parts) - 1 and isinstance(matching_child, RepetitionElement):
+            # Since each sub-element of a RepetitionElement may be rendered many times, a reference into it would be
+            # ambiguous.
+            raise FormError(referee, f"Cannot reference repeated element '{reference}' from the outside")
+
+        parents.append(matching_child)
+
+    # Reference points to parents[-1].
+    return parents[-1]
+
+
+def _get_children(node: _FormNode) -> Sequence[_FormNode]:
+    if isinstance(node, OptionsFormDefinition):
+        return *node.general, *node.sections
+
+    if isinstance(node, (FormSection, GroupElement, RepetitionElement)):
+        return node.elements
+
+    # Node is a leaf, i.e. can't have children.
+    return []
+
+
+_valid_referents: dict[type, tuple[type, ...]] = {
+    IsChecked: (CheckboxElement,),
+    IsNotChecked: (CheckboxElement,),
+    Equals: (TextInputElement, CheckboxElement, RadioGroupElement, SelectElement, HiddenElement),
+    DoesNotEqual: (TextInputElement, CheckboxElement, RadioGroupElement, SelectElement, HiddenElement),
+    In: (TextInputElement, CheckboxElement, RadioGroupElement, SelectElement, HiddenElement)
+}
+
+
+def _validate_node(node: _FormNode, parents: Sequence[_FormNode]) -> None:
+    if isinstance(node, CanHaveConditions):
+        for condition in chain(node.disable_if, node.hide_if):
+            abs_name = _absolute_name(*parents, node)
```

Comment:
`abs_name` müsste hier auch außerhalb der Schleife berechnet werden können.

```suggestion
    abs_name = _absolute_name(*parents, node)
        for condition in chain(node.disable_if, node.hide_if):
```

Reply:
Good point. Ich habe noch ein `and (node.disable_if or node.hide_if)` davor gesetzt, dann wird es auch für Elemente ohne conditions nicht ausgeführt ^^